Add generic pebble bed tutorial (#271)
idaholab/virtual_test_bed: doc/content/htgr/generic-pbr-tutorial/step5.md

!listing htgr/generic-pbr-tutorial/step5.i block=drag_bottom_reflector

The `multipliers` parameter makes the pressure drop in the x-direction much
larger than in the y-direction modeling a pipe oriented along
the y-direction.

Some materials based on correlations pick up a characteristic length as parameter `characteristic_length`. The characteristic length is usually different in different
parts of the model so it is good practice to set it using `PiecewiseByBlockFunctorMaterial`.

!listing htgr/generic-pbr-tutorial/step5.i block=characteristic_length

Currently, we only set the characteristic length in the pebble
bed and bottom reflector because that is where we use empirical correlations.

## Postprocessors

The appropriate way to get an average outlet temperature is to weight it by the mass flux and then integrate over the outlet boundary. This is accomplished in MOOSE using the `MassFluxWeightedFlowRate` object:

!listing htgr/generic-pbr-tutorial/step5.i block=mass_flux_weighted_Tf_out

It works very similar to the already discussed `VolumetricFlowRate` postprocessor objects that are used to compute mass flow rate and enthalpy flow rates.

> ## Executable

Reviewer: ```suggestion
## Execution
```

!listing
./pronghorn-opt -i step5.i

## Results

> We obtain an average outlet temperature of $1132$ K which is

Reviewer: finish the sentence

!media generic-pbr-tutorial/MeshP5.png
    style=width:20%
    id=step5mesh
    caption= Mesh for Step 5.

!media generic-pbr-tutorial/T_fluidP5.png
    style=width:20%
    id=step5T_fluid

>     caption= Tempurature of the fluid for Step 5.

Reviewer: ```suggestion
    caption= Temperature of the fluid for Step 5.
```

!media generic-pbr-tutorial/PressureP5.png
    style=width:20%
    id=step5Pressure
    caption= Pressure of the system for Step 5.

!media generic-pbr-tutorial/VelocityP5.png
    style=width:20%
    id=step5Velocity

>     caption= Velocity of the system for Step 5.

Reviewer: ```suggestion
    caption= Fluid velocity in the system for Step 5.
```

!media generic-pbr-tutorial/T_solidP5.png
    style=width:20%
    id=step5T_solid

>     caption= Tempurature of the solid for Step 5.    

Reviewer: ```suggestion
    caption= Temperature of the solid for Step 5.    
```
